scipy/signal/_splinemodule.cc

```diff
@@ -85,39 +90,38 @@ static PyObject *FIRsepsym2d(PyObject *NPY_UNUSED(dummy), PyObject *args)
 
   switch (thetype) {
   case NPY_FLOAT:
-    ret = S_separable_2Dconvolve_mirror((float *)PyArray_DATA(a_image),
+
+    ret = _separable_2Dconvolve_mirror((float *)PyArray_DATA(a_image),
 					(float *)PyArray_DATA(out), M, N,
 					(float *)PyArray_DATA(a_hrow),
 					(float *)PyArray_DATA(a_hcol),
 					PyArray_DIMS(a_hrow)[0], PyArray_DIMS(a_hcol)[0],
 					instrides, outstrides);
     break;
   case NPY_DOUBLE:
-    ret = D_separable_2Dconvolve_mirror((double *)PyArray_DATA(a_image),
+    ret = _separable_2Dconvolve_mirror((double *)PyArray_DATA(a_image),
 					(double *)PyArray_DATA(out), M, N,
 					(double *)PyArray_DATA(a_hrow),
 					(double *)PyArray_DATA(a_hcol),
 					PyArray_DIMS(a_hrow)[0], PyArray_DIMS(a_hcol)[0],
 					instrides, outstrides);
     break;
-#ifdef __GNUC__
   case NPY_CFLOAT:
-    ret = C_separable_2Dconvolve_mirror((__complex__ float *)PyArray_DATA(a_image),
-					(__complex__ float *)PyArray_DATA(out), M, N,
-					(__complex__ float *)PyArray_DATA(a_hrow),
-					(__complex__ float *)PyArray_DATA(a_hcol),
+    ret = _separable_2Dconvolve_mirror(reinterpret_cast<std::complex<float> *>(PyArray_DATA(a_image)),
+					reinterpret_cast<std::complex<float> *>(PyArray_DATA(out)), M, N,
+					reinterpret_cast<std::complex<float> *>(PyArray_DATA(a_hrow)),
+					reinterpret_cast<std::complex<float> *>(PyArray_DATA(a_hcol)),
```

Comment:
This is meat of the matter, parroted from scipy.special and suggested by @steppi (thank you Albert)